spec/rspec/mocks/matchers/have_received_spec.rb

```diff
@@ -459,6 +459,73 @@ module Mocks
             }.to raise_error(/received :two out of order/m)
           end
 
+          context "Specifying ordering for calls that are interleaved with each other" do
+            it 'passes when using `have_received`' do
```

Comment:
This docstring suggests that it always passes when using `have_received`, which is not true.  (Likewise for the similar docstring for `receive` below).  When stating something like this, the doc string should state under what conditions it passes since it does not always pass.

Instead, WDYT about this structure:

``` ruby
context "Specifying ordering for calls that are interleaved with each other" do
  it "passes when the actual and expected orderings match"
  it "fails when the actual and expected orderings do not match"
end
```

Notice also that we're not mentioning `have_received` in the doc string -- this file is `have_received_spec.rb` and is entirely about `have_received` so there's no need to mention it.  The `receive` examples you added below should find another home (probably `mock_ordering_spec.rb`).